openbook_auth/models.py

```diff
@@ -1445,23 +1445,23 @@ def get_community_post_with_id(self, post_id):
 
         post_query.add(~Q(community__banned_users__id=self.pk), Q.AND)
 
+        # Check we have membership or its a public community
         post_query_visibility_query = Q(community__memberships__user__id=self.pk)
         post_query_visibility_query.add(Q(community__type=Community.COMMUNITY_TYPE_PUBLIC, ), Q.OR)
-        post_query_visibility_query = Q(community__memberships__user__id=self.pk, is_closed=False)
 
-        post_query_visibility_query.add(Q(community__memberships__user__id=self.pk,
-                                          community__memberships__is_administrator=True,
-                                          is_closed=True), Q.OR)
-        post_query_visibility_query.add(Q(community__memberships__user__id=self.pk,
-                                          community__memberships__is_moderator=True,
-                                          is_closed=True), Q.OR)
-        post_query_visibility_query.add(Q(community__memberships__user__id=self.pk,
-                                          creator=self,
-                                          is_closed=True), Q.OR)
+        post_query.add(post_query_visibility_query, Q.AND)
 
-        post_query_visibility_query.add(Q(community__type=Community.COMMUNITY_TYPE_PUBLIC, is_closed=False), Q.OR)
+        # Check its not closed or we are community staff
+        post_query_closed_query = Q(is_closed=False)
```

Comment:
a bit confusing, i assumed `post_query_closed_query` filters out closed posts, but its the opposite. 
perhaps, `post_query_open_query`?

Reply:
Well it filters out the closed posts if not a staff member 🤔 , anyway this is gone on my next commit where I merged my blocked changes